internal/provider/unit.go

```diff
@@ -108,8 +108,8 @@ func (p *p) toContainers(stats map[string]*unit.State) ([]corev1.Container, []co
 		s := stats[k]
 		u, _ := unit.NewFile(s.UnitData)
 		container := v1.Container{
```

Comment:
String splitting and validation could happen here and it would only be done once instead of per call to `Name` and `Image`.

Reply:
this unit name splitting is done in various places, so I don't understand how this would help with that?

Reply:
Maybe the context of the PR is not enough and it requires me to go through the code and see where this is called. So I'll take a step back on this one.